js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -234,16 +235,51 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
     async sendTransaction(
         transaction: Transaction,
         connection: Connection,
-        _options?: SendOptions,
+        options?: SendOptions,
     ): Promise<TransactionSignature> {
         return await this.runWithGuard(async () => {
             const { authToken } = this.assertIsAuthorized();
             try {
                 return await this.transact(async (wallet) => {
-                    await this.performReauthorization(wallet, authToken);
+                    let targetCommitment: Finality;
+                    switch (connection.commitment) {
+                        case 'confirmed':
+                        case 'finalized':
+                        case 'processed':
+                            targetCommitment = connection.commitment;
+                            break;
+                        default:
+                            targetCommitment = 'finalized';
+                    }
+                    await Promise.all([
+                        this.performReauthorization(wallet, authToken),
+                        (async () => {
+                            if (transaction.recentBlockhash == null) {
+                                const preflightCommitment = options?.preflightCommitment;
+                                const preflightCommitmentScore =
+                                    preflightCommitment === 'finalized'
+                                        ? 2
+                                        : preflightCommitment === 'confirmed'
+                                        ? 1
+                                        : 0;
+                                const targetCommitmentScore =
+                                    targetCommitment === 'finalized' ? 2 : targetCommitment === 'confirmed' ? 1 : 0;
+                                const { blockhash } = await connection.getLatestBlockhash({
+                                    commitment:
+                                        preflightCommitment && preflightCommitmentScore < targetCommitmentScore
+                                            ? preflightCommitment
+                                            : targetCommitment,
+                                });
```

Comment:
Assuming `preflightCommitment` and `targetCommitment` are the same value (reasonable default) or `preflightCommitment` isn't provided (also reasonable) could just check for truthy equality first and prefer `targetCommitment`

Reply:
Is it not possible to have shenanigans like `preflightCommitment` being `processed` but `targetCommitment` being a higher commitment like `finalized`? That wouldn't work with the scheme above, right?